src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
+                if (row['#gene1'] == gene1 or row['#gene1'] == gene2) and (
+                        row['gene2'] == gene1 or row['gene2'] == gene2):
+                    breakpoint1 = row['breakpoint1']
+                    breakpoint2 = row['breakpoint2']
+                    break
+
+        if not (breakpoint1 and breakpoint2):
+            raise ValueError(f"No matching fusion found in the TSV file for {fusion}.")
+
+        # Function to modify the breakpoint format to "chr:start-end"
+        def format_breakpoint(breakpoint):
+            chrom, pos = breakpoint.split(':')
+            start = int(pos)
+            end = start + 1
+            return f"{chrom}:{start}-{end}"
+
+        formatted_breakpoint1 = format_breakpoint(breakpoint1)
+        formatted_breakpoint2 = format_breakpoint(breakpoint2)
+
+        # Load the JSON template
+        with open(json_template_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        # Update the JSON with the formatted breakpoints
+        data['locus'] = [formatted_breakpoint1, formatted_breakpoint2]
+
+        tumour_id = wrapper.get_my_string(core_constants.TUMOUR_ID)
+        data['tracks'][1]['name'] = tumour_id
+
+        # Search for the BAM and BAI files using glob.glob
+        bam_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bam"
+        bai_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bai"
+
+        bam_files = glob.glob(bam_pattern)
+        bai_files = glob.glob(bai_pattern)
+
+        if bam_files:
+            bam_file = bam_files[0]
+            project = os.path.basename(os.path.dirname(os.path.dirname(bam_file)))
+            filename = os.path.basename(bam_file)
+            data['tracks'][1]['url'] = f"/bams/project/{project}/RNASEQ/file/{filename}"
+        else:
+            raise FileNotFoundError(f"BAM file not found for pattern: {bam_pattern}")
+
+        if bai_files:
+            bai_file = bai_files[0]
+            project = os.path.basename(os.path.dirname(os.path.dirname(bai_file)))
+            filename = os.path.basename(bai_file)
+            data['tracks'][1]['indexURL'] = f"/bams/project/{project}/RNASEQ/file/{filename}"
+        else:
+            raise FileNotFoundError(f"BAI file not found for pattern: {bai_pattern}")
+
+        # Write the modified JSON to the output directory
+        output_json_path = os.path.join(output_dir, f"{fusion}.json")
+        with open(output_json_path, 'w') as json_output_file:
+            json.dump(data, json_output_file)
+
+        with open(output_json_path, 'r') as json_output_file:
+            json_content = json_output_file.read()
+        compressed_b64_data = self.compress_string(json_content)
+        blurb_url = f"https://whizbam.oicr.on.ca/igv?sessionURL=blob:{compressed_b64_data}"
+        return fusion, blurb_url
+
+    def compress_string(self, input_string):
+        # Convert string to bytes
+        input_bytes = input_string.encode('utf-8')
```

Comment:
Please use the value from core constants instead of hard-coding: https://github.com/oicr-gsi/djerba/blob/24bba4e4f9f1b1e2db039a2d688e7ff677afde4d/src/lib/djerba/core/constants.py#L4

Reply:
Why has this been marked resolved?

Reply:
I've update it here 
 ```
def compress_string(self, input_string):
        # Convert string to bytes
        input_bytes = input_string.encode(core_constants.TEXT_ENCODING)
```